src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -16,46 +16,76 @@
+internal sealed class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
-    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
 
-    static ProcessMetrics()
+    private static readonly ThreadLocal<InstrumentsValues> CurrentThreadInstrumentsValues = new(() => new InstrumentsValues());
+
+    public ProcessMetrics(ProcessInstrumentationOptions options)
     {
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.usage",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
-            },
+            () => CurrentThreadInstrumentsValues.Value.GetMemoryUsage(),
             unit: "By",
             description: "The amount of physical memory in use.");
 
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.virtual",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.VirtualMemorySize64;
-            },
+            () => CurrentThreadInstrumentsValues.Value.GetVirtualMemoryUsage(),
             unit: "By",
             description: "The amount of committed virtual memory.");
     }
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="ProcessMetrics"/> class.
-    /// </summary>
-    /// <param name="options">The options to define the metrics.</param>
-    public ProcessMetrics(ProcessInstrumentationOptions options)
+    private sealed class InstrumentsValues
     {
+        private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
+        private double? memoryUsage;
+        private double? virtualMemoryUsage;
+
+        internal InstrumentsValues()
+        {
+            this.memoryUsage = null;
+            this.virtualMemoryUsage = null;
+        }
+
+        internal double GetMemoryUsage()
+        {
+            if (!this.memoryUsage.HasValue)
+            {
+                this.Snapshot();
+            }
+
+            var value = (double)this.memoryUsage;
+            this.memoryUsage = null;
+            return value;
+        }
+
+        internal double GetVirtualMemoryUsage()
+        {
+            if (!this.virtualMemoryUsage.HasValue)
+            {
+                this.Snapshot();
+            }
+
+            var value = (double)this.virtualMemoryUsage;
+            this.virtualMemoryUsage = null;
+            return value;
+        }
+
+        private void Snapshot()
+        {
+            CurrentProcess.Refresh();
+            this.memoryUsage = CurrentProcess.WorkingSet64;
+            this.virtualMemoryUsage = CurrentProcess.PagedMemorySize64;
```

Comment:
Did you intend to use `CurrentProcess.VirtualMemorySize64` here?

Reply:
Nope, this update was because I found out that PagedMemorySize64 should be the more suitable value for this gauge.
Please refer to this [PR](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/673) for my rationale.